### PR TITLE
457 generate audit logs plus new helper interfaces

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Helpers for creating audit logs in Logger class (made the imports easier)
- More helpers added to abstract how it's implemented to the calling code 
- Much more useage of Audit logging added

---

**REST API Updates **

The /api/v1/audit_logs GET now takes an optional 'module_name' parameter which means that it returns only the audit log options for that module

### CODE REVIEW

File: app/modules/audit_logs/models.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 import uuid
 
+AuditMaxMessageLength = 240
+
```
Comment (colinwkingen) [inline app/modules/audit_logs/models.py:15]: I've been thinking a lot about the message length. From the EDM, java stack traces can be enormous, but useful. The exception messages we return from the EDM are pretty small right now (and less useful). We might want to store the longer ones.

**Review by colinwkingen — APPROVED** [review]
LGTM. This is pretty cool stuff, I'm looking forward to what we can do with it in the future.

File: app/modules/audit_logs/models.py (hunk 1 above)
Comment (hwindsor) [reply app/modules/audit_logs/models.py:15]: Agreed. This length limitation is going to be removed as discussed yesterday

**Pull request merged**